### DISCUSSION TITLE
Consent prompting

### DISCUSSION THREAD

**Discussion by aklidata** [PullDating/Org#82] [Ideas & Features]
consent prompts: basically if conversation starts to turn sexual app will prompt both parties to consent or reject the conversation. If one rejects and the AI detects one party still trying to push for sexual content they get penalized

**Comment by ajbott** [comment]
I think instead of doing this mid conversation, the consent would happen before the conversation starts. The person would set their conversation boundaries on their profile so that when they match, the other party would have to agree to respect their boundaries in the conversation. If they don't, then this would give us grounds to punish them for not respecting their boundaries.

**Comment by flo-kei** [comment]
I'm not sure if analyzing conversations is going to earn us the trust of the users.
People are used to end-to-end encryption of other apps (whatsapp, signal, etc), so why would they use the chat feature of one that looks at every word you speak and potentially censors you based on this 

maybe a report-feature could take care of this, in combination with the idea that @ajbott suggested: specifying their boundaries beforehand (and maybe also displaying this to both parties somewhere?, so they both know what terms they are on..)

**Reply by ajbott** [reply]
I had the idea of maybe adding a "allow moderators to view chat" feature where it would require at least 1 member of that chat to give consent for moderators to view the chat. When a user reports someone for breaking the consent prompt, moderators could view the chat to determine if this is the case and make a ruling based on the information provided.

As far as AI analysis, I don't think it would be smart enough to distinguish consensual conversation.

**Reply by flo-kei** [reply]
This sounds good to me 👍🏻 

I'm still not 100% happy with the fact that a moderator could be listening in on a private conversation at any point, but I guess some sort of moderation makes sense, given that inappropriate behaviour is and will be part of the experience :/

Maybe we should clarify this beforehand somewhere, that when a report is sent, that chat becomes visible to moderators. (maybe a disclaimer in some document?)

**Comment by Oshyan** [comment]
I'm not sure this is the best solution to the problem of unwanted sexual content, but it's certainly worth thinking about the options.

How does this really differ from and have advantage over simply Reporting a conversation? One of the Report options could just be "Unwanted sexual advances" or whatever. We can treat different reported issues with different levels of severity and address them accordingly. 

So I'm not sure there's a need for a whole consent-dedicated system, or if there is I don't think it should focus just on sexual content, and perhaps starts to sound more like a mutual compatibility consideration that could either be dealt with in matching/filtering, or is more appropriate offline for those people to address between them.
